src/Simulation/QSharpFoundation/Convert/Convert.qs

```diff
@@ -63,29 +63,76 @@ namespace Microsoft.Quantum.Convert {
     /// Converts a given big integer to an equivalent integer, if possible.
     /// The function returns a pair of the resulting integer and a Boolean flag
     /// which is true, if and only if the conversion was possible.
-    /// # Remarks
-    /// See [C# BigInteger constructor](https://docs.microsoft.com/dotnet/api/system.numerics.biginteger.-ctor?view=netframework-4.7.2#System_Numerics_BigInteger__ctor_System_Int64_) for more details.
     function MaybeBigIntAsInt(a : BigInt) : (Int, Bool) {
-        body intrinsic;
+        if a > (1L <<< 63) - 1L or a < (-1L * (1L <<< 63)) {
+            return (0, false);
+        }
```

Comment:
I'm not sure the fragment `(1L <<< 63)` gives the value you expect. I expect that it causes the overflow of the 64-bit signed type. Would be nice to have tests for the cases `a > ..` and `a < ..`.

Reply:
If we have those tests then OK (and I assume that they succeed for QIR RT also, not only for C# RT). 

I was afraid that the fragment `(1L <<< 63)` could be calculated as a signed 64-bit integer (rather than `BigInt`), resulting in a value 0x8000_0000_0000_0000 (the lowest negative 64-bit signed value). Then the fragment `-1L * ..` could also be calculated as signed 64-bit integer (rather than `BigInt`), resulting in negating the lowest negative value, which could overflow the signed 64-bit integer. And then the overflown signed 64-bit value could be compared to BigInt.

Reply:
Yeah, the interesting thing to recognize here is that bitshift of a `BigInt` is calculated through on operation that takes a `BigInt` target to be shifted and an `Int` number of bits to shift, and the calculation takes place in a `BigInt` with all the fancy automatic bit width adjustments. As a result, the numbers work as expected:
`(1L <<< 63) - 1L` becomes `9223372036854775807`
`(-1L * (1L <<< 63))` becomes `-9223372036854775808`
which are 64-bit signed int max and min, respectively.

Reply:
No worries, I probably should've mentioned that earlier! Always good to double check ;)